app/models/concerns/foreman_bootdisk/host_ext.rb

```diff
@@ -14,6 +14,7 @@ def bootdisk_template_render
 
   def bootdisk_chain_url(mac = self.mac, action = 'iPXE')
     u = URI.parse(foreman_url(action))
+    raise(::Foreman::Exception.new(N_('Foreman or proxy is configured with https, not supported for iPXE'))) if u.scheme == 'https'
```

Comment:
Isn't this very particular to the build of iPXE?  It seems a little limiting to me if it was rebuilt with the appropriate CA.